tmt/cli.py

```diff
@@ -2030,20 +2030,21 @@ def setup_completion(shell: str, install: bool) -> None:
     else:
         script = Path(config.path) / f'{COMPLETE_SCRIPT}.{shell}'
 
-    # SIM115: Use context handler for opening files. Would not reduce complexity here.
-    out = open(script, 'w') if install else sys.stdout  # noqa: SIM115
-    subprocess.run(f'{COMPLETE_VARIABLE}={shell}_source tmt',
-                   shell=True, stdout=out)
+    command = f'{COMPLETE_VARIABLE}={shell}_source tmt'
 
     if install:
-        out.close()
+        with open(script, 'w') as f:
+            subprocess.run(command, stdout=f)
```

Comment:
`shell=True` was dropped, what seems like a bug. I am all in dropping it, but then, we need to use `env` to pass the variable into the environment.

Reply:
Thanks. Adding it back, as you mentioned, we would need env.
Do we want to disable the [S602](https://docs.astral.sh/ruff/rules/subprocess-popen-with-shell-equals-true/) completely?

Reply:
Addressed in https://github.com/teemtee/tmt/pull/2842/commits/706c670a649614f3d616e80b39120af29202101f

Let me know if `noqa` would be preferred, so ruff checks for `shell=True` going forward

Reply:
I think in general, when possible, we should avoid using it, as it can open some holes in theory.
In this case, seems the only thing we need is adding another environment variable, in that case I would rather go with `env`, copy the process environment and add the one we need.

@WDYT @teemtee/tmt

Reply:
Dropping `shell` does sound like a good idea. If all we need is to add an environment variable to the mix, then `env` is exactly the answer.

Reply:
+1 to thrix' proposal.

S602 is good check to  have however in tmt we are expected to run as shell sometimes. But we need to be aware that it is the case and, maybe, over time find way to avoid doing so ;)

Reply:
Had to look at the implications of `shell=True`, and from what I understand it is to make sure a single command is being run
```python
# insecure
some_insecure_flags="-la; do_some_weird_stuff"
subprocess.run(f"ls {some_insecure_flags}", shell=True)
# more secure
sanitized_flags=shlex.split(some_insecure_flags)
subprocess.run(["ls", *sanitized_flags])
```
But I think in the case for tmt, we generally want to enable the "insecure" approach in order to run a yaml text block, but probably this can be scoped specifically to running the test scripts parts. As for `env`, I am not sure, you mean the part about prefixing `f"{COMPLETE_VARIABLE}={shell}_source"`?

Reply:
Yep, sometimes tmt needs to run commands with `shell=True`, but this does not seem to be the case. `f'{COMPLETE_VARIABLE}={shell}_source tmt'` - no redirection, no shell magic, no flow controls, just a command with environment spiked with a single extra variable. `Command('tmt').run(env={COMPLETE_VARIABLE: f'{shell}_source'})`, done :)

Reply:
> `Command('tmt').run(env={COMPLETE_VARIABLE: f'{shell}_source'})`, done :)

oh, that's way better. Doing that.

Reply:
Yep, I for one would go with `tmt.Command` (or `tmt.utils.Command`) as the way to run external commands in the tmt codebase.

Reply:
LGTM.